From 418b18f5162f621aa9848fe1eaaaf36f855c6d81 Mon Sep 17 00:00:00 2001
From: Liu Xiaowen <b37945@freescale.com>
Date: Thu, 15 Sep 2011 15:56:58 +0800
Subject: [PATCH 180/502] ENGR00156721 add yuv 420 planar format for gralloc

add yuv 420 planar format for gralloc to support MJPEG video.

Signed-off-by: Liu Xiaowen <b37945@freescale.com>
---
 mx5x/libgralloc/gralloc.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mx5x/libgralloc/gralloc.cpp b/mx5x/libgralloc/gralloc.cpp
index 0c40ae4..bc66115 100755
--- a/mx5x/libgralloc/gralloc.cpp
+++ b/mx5x/libgralloc/gralloc.cpp
@@ -367,12 +367,13 @@ static int gralloc_alloc(alloc_device_t* dev,
 
     size_t size, alignedw, alignedh;
     if (format == HAL_PIXEL_FORMAT_YCbCr_420_SP || 
-            format == HAL_PIXEL_FORMAT_YCbCr_422_SP) 
+            format == HAL_PIXEL_FORMAT_YCbCr_422_SP || format == HAL_PIXEL_FORMAT_YCbCr_420_I) 
     {
         // FIXME: there is no way to return the alignedh
         alignedw = (w + 1) & ~1; 
         switch (format) {
             case HAL_PIXEL_FORMAT_YCbCr_420_SP:
+            case HAL_PIXEL_FORMAT_YCbCr_420_I:
                 size = alignedw * h * 2;
                 break;
             case HAL_PIXEL_FORMAT_YCbCr_422_SP:
-- 
1.8.0

